### PR TITLE
Profile page

### PR DESCRIPTION
**Pull request by shamilclm**

### CODE REVIEW

File: lib/core/widgets/app_bar_widget.dart [hunk 1]
```diff
@@ -1,48 +1,36 @@
...
+  Widget build(BuildContext context) {
+    final appTheme = AppTheme.of(context);
+    return AppBar(
+      scrolledUnderElevation: 0,
+      titleSpacing: appTheme.spaces.space_300,
+      title: Row(
+        children: [
+          InkWell(
+            child: SvgPicture.asset(
+              'assets/icons/ic_arrow_backward.svg',
```
Comment (thadathilsinan) [inline lib/core/widgets/app_bar_widget.dart]: put the asset into constants

File: lib/features/profile_page/presentation/pages/profile_page.dart [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final apptheme = AppTheme.of(context);
+    return Scaffold(
+      appBar: AppBarWidget(
+        title: ref.watch(profilePageProvider).txtTitle,
+      ),
+      body: Padding(
+        padding: const EdgeInsets.symmetric(horizontal: 24, vertical: 32),
```
Comment (thadathilsinan) [inline lib/features/profile_page/presentation/pages/profile_page.dart]: use spaces

File: lib/features/profile_page/presentation/pages/profile_page.dart [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+              alignment: Alignment.center,
+              child: CircleAvatar(
+                radius: apptheme.spaces.space_200 * 7,
+              ),
+            ),
+            SizedBox(
+              height: apptheme.spaces.space_400,
+            ),
+            Text(
+              ref.watch(profilePageProvider).txtOpeningTime,
```
Comment (thadathilsinan) [inline lib/features/profile_page/presentation/pages/profile_page.dart]: make a final variable on top and use it everywhere

File: lib/features/profile_page/presentation/providers/switch_button_provider.dart [hunk 4]
```diff
@@ -0,0 +1,3 @@
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+final switchprovider = StateProvider<bool>((ref) => false);
```
Comment (thadathilsinan) [inline lib/features/profile_page/presentation/providers/switch_button_provider.dart]: use camelcase

Comment (thadathilsinan) [inline lib/features/profile_page/presentation/providers/switch_button_provider.dart]: create a notifier provider 'profileProvider'

File: pubspec.lock
Comment (thadathilsinan) [inline pubspec.lock:1]: Remove this file
git rm pubspec.lock

File: pubspec.yaml [hunk 5]
```diff
@@ -92,7 +58,36 @@ flutter:
         - asset: assets/fonts/Inter/Inter-Black.ttf
           weight: 900
 
+
+
```
Comment (thadathilsinan) [inline pubspec.yaml]: remove alll blank lines

**Review by thadathilsinan — CHANGES_REQUESTED** [review]

File: lib/core/constants/profile_page/profile_page_constants.dart [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+class ProfilepageConstants {
+  final txtTitle = 'Profile';
+
+  final txtOpeningTime = 'Opening time';
+  final txtClosingtime = 'Closing time';
+  final txtDarktheme = 'Dark theme';
+  final txtUpdatePassword = 'Update Password';
+  final txtEdit = 'Edit';
+  final icArrowBackward = 'assets/icons/ic_arrow_backward.svg';
```
Comment (thadathilsinan) [inline lib/core/constants/profile_page/profile_page_constants.dart:12]: Add into the assets constants file

File: lib/core/constants/profile_page/profile_page_constants.dart [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+part 'profile_page_constants.g.dart';
+
+class ProfilepageConstants {
```
Comment (thadathilsinan) [inline lib/core/constants/profile_page/profile_page_constants.dart]: Check the case

File: lib/core/widgets/add_image_widget.dart
Comment (thadathilsinan) [inline lib/core/widgets/add_image_widget.dart:1]: Uncomment

File: lib/core/widgets/app_bar_widget.dart [hunk 8]
```diff
@@ -1,48 +1,38 @@
...
-//   @override
-//   // TODO: implement preferredSize
-//   Size get preferredSize => throw UnimplementedError();
-
-//   @override
-//   // TODO: implement child
-//   Widget get child => throw UnimplementedError();
-// }
+  @override
+  Size get preferredSize => const Size.fromHeight(56);
```
Comment (thadathilsinan) [inline lib/core/widgets/app_bar_widget.dart:38]: use spaces

File: pubspec.lock
Comment (thadathilsinan) [inline pubspec.lock:1]: Remove pubspec lock

git rm pubspec.lock

**Review by thadathilsinan — CHANGES_REQUESTED** [review]

File: .gitignore
Comment (thadathilsinan) [inline .gitignore:1]: reset this file

**Review by thadathilsinan — CHANGES_REQUESTED** [review]

File: lib/core/widgets/add_image_widget.dart
Comment (shamilclm) [reply lib/core/widgets/add_image_widget.dart:1]: fixed

File: lib/core/widgets/app_bar_widget.dart (hunk 8 above)
Comment (shamilclm) [reply lib/core/widgets/app_bar_widget.dart:38]: fixed

**Review by thadathilsinan — APPROVED** [review]

**Pull request merged**